File: galaxy/agent/models/common_result.go
package models

type CommonResult struct {
	Code    MCode       `json:"code"`
	Message string      `json:"message"`
	Data    interface{} `json:"data"`
}

func NewCommonResult(code MCode, message string, data interface{}) *CommonResult {
	return &CommonResult{Code: code, Message: message, Data: data}
}

func ParseCommonResult(data interface{}) *CommonResult {
	if data == nil {
		return nil
	}
	if result, ok := data.(*CommonResult); ok {
		return result
	}
	return nil
}

func (r *CommonResult) SetCode(code MCode) *CommonResult {
	r.Code = code
	return r
}

func (r *CommonResult) SetMessage(message string) *CommonResult {
	r.Message = message
	return r
}

File: quantum/dotnet/molecule/Molecule/Models/CommonResult.cs

namespace Molecule.Models
{
    public class CommonResult<T>
    {
        public Codes Code { get; init; } = Codes.Ok;
        public string? Message { get; init; } = "";
        public T? Data { get; init; } = default(T);
    }

    public class SelectData<T>
    {
        public int Count { get; init; } = 0;
        public List<T> List { get; init; } = new List<T>();
    }

    public class SelectResult<T> : CommonResult<SelectData<T>> { }
}